rfcs/0002-authentication.md

```diff
@@ -1,12 +1,12 @@
 ---
-RFC: 0002
+RFC: 2
```

Comment:
I've changed the RFC numbers to not include the preceding 000's, it's easier to read without them.